staticcheck/lint.go

```diff
@@ -1187,6 +1188,29 @@ func (c *Checker) CheckDiffSizeComparison(j *lint.Job) {
 	}
 }
 
+func (c *Checker) CheckTimeEqualityComparison(j *lint.Job) {
+	for _, ssafn := range j.Program.InitialFunctions {
+		for _, block := range ssafn.Blocks {
+			for _, ins := range block.Instrs {
+				binop, ok := ins.(*ssa.BinOp)
+				if !ok {
+					continue
+				}
+				if binop.Op != token.EQL && binop.Op != token.NEQ {
+					continue
+				}
+
+				if types.TypeString(binop.X.Type(), nil) != "time.Time" ||
+					types.TypeString(binop.Y.Type(), nil) != "time.Time" {
+					continue
+				}
+
+				j.Errorf(binop, "using == or != operators to compare time is problematic; prefer Time.Equal")
```

Comment:
Reword the message to read `should use time.Time.Equal when comparing time.Time values`